core/capi/jml.h

```diff
@@ -0,0 +1,53 @@
+/**
+ * This file provides the C API to JML.
+ * Author: Adam Hutchings
+ * Date: 10-07-23
+ */
+
+#ifndef JML_API_H
+#define JML_API_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+/**
+ * This adds the entries in v2 to v1.
+ */
+int jml_vector_add(unsigned v1, unsigned v2);
+
+/**
+ * This adds val to the pos-th entry in the vector vid.
+ */
+int jml_vector_add_entry(unsigned vid, int pos, double val);
+
+enum JMLActivFunc {
+    JACT_FSIG,
+    JACT_SIG,
+    JACT_RELU,
+};
```

Comment:
nit: Use typedef to make code less clunky
```suggestion
typedef enum JMLActivFunc {
    JACT_FSIG,
    JACT_SIG,
    JACT_RELU,
} JMLActivFunc;
```
I would note that a more verbose name is better, but I suppose this is alright.